File: AlgoExpert/TwoSum.py
def twoNumberSum(array, targetSum):
    dictionary = {}
    for i in (array):
        num = targetSum - i
        if num in dictionary:
            return [num, i]
        else:
            dictionary[i] = num
    return []
print(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 10))

File: AlgoExpert/Two Sum/TwoSum.cs
using System;

public class Program {
	public static int[] TwoNumberSum(int[] array, int targetSum) {
		// Two Pointer approach
		Array.Sort(array);
		int left = 0;
		int right = array.Length - 1;
		while(left < right)
		{
			int currentSum = array[left] + array[right];
			if(currentSum == targetSum)
			{
				return new int[] {array[left], array[right]};
			}
			if(currentSum > targetSum)
			{
				right --;
			}
			else if(currentSum < targetSum)
			{
				left ++;
			}
		}
		return new int[0];
	}
}
